common/lib/dependabot/update_checkers/version_filters.rb

```diff
@@ -9,11 +9,20 @@ module VersionFilters
       extend T::Sig
 
       sig do
-        params(
-          versions_array: T::Array[T.any(Gem::Version, T::Hash[Symbol, Gem::Version])],
-          security_advisories: T::Array[SecurityAdvisory]
-        )
-          .returns(T::Array[T.any(Gem::Version, T::Hash[Symbol, Gem::Version])])
+        # Tricky generics explanation:
+        # There's a type T that is either a Gem::Version or a Hash with a :version key
+        # The method returns an array of T
+        # So whichever is provided as input, the output will be an array of the same type.
```

Comment:
nit: add a link to the documentation here https://sorbet.org/docs/generics#placing-bounds-on-generic-methods